assets/src/blocks/CarouselHeader/Sidebar.js

```diff
@@ -27,13 +49,58 @@ export const Sidebar = ({
         onChange={value => setAttributes({ carousel_autoplay: value })}
       />
     </PanelBody>
-    {slides.map((slide, index) => (
+    { slides.map((slide, index) => <div
+      style={ {
+        // When you start dragging an element, what is shown while dragging is a bitmap snapshot of what the element
+        // looked like the moment the drag started. The .01s delay should be plenty to ensure that snapshot is taken
+        // before we hide the element.
+        transform: index === draggedSlide ? 'scaleY(0)' : 'none',
+        transitionDelay: '.01s',
+        transitionProperty: 'transform',
+      } }
+      className={ index === dragTarget ? 'carousel-sidebar-insert-before' : 'carousel-sidebar-slide' }
+      key={ `${ slide.image_srcset }${ index }` }
+      draggable
+      onDragOver={ () => {
+        timeoutRef.current && window.clearTimeout(timeoutRef.current);
+        setDragTarget(index);
+      } }
+      onDragLeave={ () => {
+        // The event fires also when going over descendant elements, then immediately fires on the parent again.
+        // Without this timeout it continuously set and unset the drag target.
+        timeoutRef.current = window.setTimeout(() => {
+          setDragTarget(null);
+        }, 100);
+      } }
+      onDragStart={ () => {
+        setDraggedSlide(index);
+      } }
+      onDragEnd={ () => {
+        setDraggedSlide(null);
+        if (dragTarget === null) {
+          return;
+        }
+        setAttributes({ slides: reOrderSlides(slides, index, dragTarget) });
+        setDragTarget(null);
+      } }
+    >
       <PanelBody
-        key={index}
-        title={`${__('Slide', 'planet4-blocks-backend')} ${index + 1}`}
-        opened={currentSlide === index}
-        onToggle={() => {
-          if (currentSlide !== index) {
+        key={ index }
```

Comment:
Could you change this key value to another one? Because the `index` value is not recommended to use as `key` through item from lists

Reply:
Good point :+1: This is actually more a problem now slides are re-orderable.

Reply:
Actually, now that it's wrapped in a div it doesn't need a key anymore. I also noticed in some other code it has a key that is not needed, I'll remove those now.